promql/engine.go

```diff
@@ -626,6 +628,9 @@ func (ng *Engine) exec(ctx context.Context, q *query) (v parser.Value, ws annota
 	defer func() {
 		ng.queryLoggerLock.RLock()
 		if l := ng.queryLogger; l != nil {
+			logger := slog.New(l)
+			f := make([]any, 0, 16) // Probably enough up front to not need to reallocate on append.
```

Comment:
Would this be more efficient if it built a `[]slog.Attr` then called `logger.LogAttrs` ?

Reply:
It should, that is indeed the more efficient log call. I'll rework how we build up the attribute list :+1: